packages/terra-form/src/_Utilities.jsx

```diff
@@ -0,0 +1,37 @@
+const autoResizeTextareaHeight = (onChange, baseRowCount) =>
+  (e) => {
+    const textarea = e.currentTarget;
+    const lineHeight = Math.ceil(parseFloat(window.getComputedStyle(textarea).lineHeight, 0));
+
+    /* Smallest Textarea should have at least 2 rows. */
+    const minRows = baseRowCount || 2;
+    textarea.rows = minRows;
+    const rows = Math.ceil((textarea.scrollHeight - textarea.baseScrollHeight) / lineHeight);
+    textarea.rows = minRows + rows;
+
+    if (onChange) {
+      onChange(e);
+    }
+  };
+
+const setTextareaBaseHeight = onFocus =>
+  (e) => {
+    if (onFocus) {
+      onFocus(e);
+    }
+
+    const textarea = e.currentTarget;
+
+    if (!textarea.baseScrollHeight) {
+      const savedValue = textarea.value;
+      textarea.value = '';
+      textarea.baseScrollHeight = textarea.scrollHeight;
+      textarea.value = savedValue;
+      autoResizeTextareaHeight(e);
+    }
+  };
+
+export default {
```

Comment:
Could we add comments describing each function and the parameters? I'm a little confused on why setTextAreaBaseHeight affects onFocus and autoResizeTextareaHeight handles onChange.

Reply:
Sure I can do that.

Essentially, we need the focus to capture the original height of the textarea, as we will be using that to calculate what the new height is anytime on change is triggered on the textarea.

Reply:
New change. I decided to switch from onFocus to the componentDidMount method to get the baseHeight. We need this change, as UX wants to resize the textarea on page load if the text is larger than the provided space. I removed the utilities here, and added comments to the individual methods that do the resizing. https://github.com/cerner/terra-core/pull/966/commits/87ed82ca6e2a7327c13b721ecfc307f95fd1a7e4